lib/graphql/query/serial_execution/value_resolution.rb

```diff
@@ -45,7 +45,8 @@ class ListResolution < BaseResolution
           def non_null_result
             wrapped_type = field_type.of_type
             strategy_class = get_strategy_for_kind(wrapped_type.kind)
-            value.map do |item|
+            value.each_with_index.map do |item, index|
+              irep_node.index = index
```

Comment:
Mutation is generally error-prone, but I'm open to it here, since it's a hot path and not used intensively. Maybe one good thing would be to put it back to `nil` when we're done, just in case? I can't think of why it would matter though. :S 

An alternative implementation would be to define a delegator, something like 

```
inner_node = IndexedNode.new(irep_node, index)
```

where `IndexedNode` is like 

```ruby 
class IndexedNode 
  extend Forwardable 

  def initialize(irep_node, index) 
    @irep_node = irep_node 
    @index = index 
  end 

  def path 
    @irep_node.path + [@index]
  end 

  def_delegators :@irep_node, :name, :definitions, :directives, # ?? Lots of stuff 
end 
```

I find this implementation less scary, since you never mess with the original `irep_node`, but it's also less efficient. I'm happy to leave it as is and refactor to something like :point_up: that if we _actually_ have issues.